interfaces/IBrowser.h

```diff
@@ -149,6 +151,26 @@ namespace Exchange {
 
         // @brief Initiate garbage collection
         virtual uint32_t CollectGarbage() = 0;
+
+        // @brief Run javascript in main frame
+        // @param script utf8 encoded JS code string
+        virtual uint32_t RunJavaScript(const string& script) = 0;
+
+        // @brief Add user script to be executed at document start
+        // @param script utf8 encoded JS code string
+        virtual uint32_t AddUserScript(const string& script, const bool topFrameOnly) = 0;
+        // @brief Remove all user scripts
+        virtual uint32_t RemoveAllUserScripts() = 0;
+
```

Comment:
Maybe good to avoid GOD interfaces that can do it all. Why not create a IBrowserScripting Interface. Using the COMRPC QueryInterface you can than always access this and if that browser (e.g. the Espial variant we have) does not support it, you just do not implement it.
It also keeps the interface lean and mean and to the point.
IAslo suggest to keep symetrical interfaces. If you can add a USerScript, why can you not Remove one user script ?

Reply:
> Why not create a IBrowserScripting Interface. 

done.

> IAslo suggest to keep symetrical interfaces. If you can add a USerScript, why can you not Remove one user script ?

There is RemoveAllUserScripts. The idea here is that you'll only need to add scripts before loading a web app, and reset browser if you want to load different web app in the same instance.